### ISSUE TITLE
Optimize power usage

### ISSUE THREAD

**Issue by BenPru** [BenPru/luxtronik#21]
Yesterday I activated my heatpump for the heating season. And now I read out the power usage of my heatpump with ha glow.
I found out today that it has a big impact between heating "off" and "automatic" without running.
The standby power usage in heating mode "off" is about ~13 Watt.
In automatic mode the standby power usage is about 83 Watt and every 20 minutes it is ~150 Watt!
Every 20 minutes the heatpump runs in mode "pump forerun" without doing something productive!
In the past I thought this is a display bug in luxtronik but the usage show a real impact.
My luxtronik version is V2.86.0 (Novelan LD7).

Has anyone something like this?

**Comment by rhammen** [comment]
I have an Alpha-Innotec Brine/Water heatpump. I do not have a separate electricity meter, so I cannot really judge the heating=off power usage. But I do not see periodic spikes in the energy usage that I could relate to the heatpump status. Electricity usage really only goes up when the heatpump is running.

**Comment by scaronni** [comment]
I'm installing a Shelly 3EM soon on all the 3 phase connections of the heat pump to do some monitoring, it seems to me my usage is crazy high but I have no way to do proper measurements.

- Heatpump
- Boiler (heating)
- Boiler (hot water)
- Heizeinsatz (Secondary power?)

I have a constant load of ~600/700 watts for home server/computer usage, so I can't really tell if the heating pump consumes too much when idle until i've installed those.

**Comment by Crashman1983** [comment]
High standby power consumption is caused by Heatpump electronics, oil heating and circulation pump. Alpha Innotec do consume a hefty amount of energy during standby - you cannot change that.

**Comment by Kars-de-Jong** [comment]
It is caused by the circulation pump (HUP). It needs to run in order to measure the actual return temperature of the heating circuit. If this is not done the measured return temperature is not reliable (especially not right after a hot water run).

You can influence it by enabling "Pump optimization", it will then (depending on the difference between actual and wanted return temperature) either run for 5 minutes every 30 minutes, or continuously when the difference between wanted and actual return temperature is small.

Energy usage also depends on whether your circulation pump can be controlled by a PWM signal. 

On my Alpha Innotec the standby usage with heating set to off is about 11 W, when set to automatic it is 25 W when the HUP runs at 50%.

**Comment by BenPru** [comment]
@Kars-de-Jong Thanks for your explanation.

> enabling "Pump optimization"

Is this a luxtronik parameter?

> I did limit the maximum speed of the HUP to 50%

Is this also a luxtronik parameter?

**Comment by Kars-de-Jong** [comment]
@BenPru Yes.

"Pump optimization" is parameter `ID_Einst_Popt_akt` of type Bool, default is 1, "Pump optimisation time" is parameter `ID_Einst_Popt_Nachlauf_akt` of type Minutes (which doesn't exist yet), default is 180, and maximum circulation pump speed for heating is `ID_Einst_P155_PumpHeat_Max` of type Percent2, default is 100.

**Comment by Kars-de-Jong** [comment]
@Crashman1983 True, but I noticed the heat pump has a rather "conservative" ΔT between supply and return flow of the heating system of 4 K. My house is designed for a ΔT of 10 K, which can't be reached with the pump running at 100%. So I limited the maximum speed. I would much rather be able to change the ΔT of 4 K the heat pump uses, but that doesn't seem to be a user configurable parameter.

**Comment by Crashman1983** [comment]
Are you using underfloor heating? Maybe you could trie to set the circulation pump settings to Radiator. I can imagine, that this has influence to ΔT (only a guess!) because I would suspect radiators to have a higher difference between supply and return flow. 

For underfloor heating 10K difference would be really much!

**Comment by Kars-de-Jong** [comment]
Yes, underfloor heating, design temperature 30 °C, ΔT 10 K, very low flows (highest flow through a loop of 115 m is 39 l/h). If you then have a circulation pump that runs at 2000 l/h that seems a bit silly.

**Comment by BenPru** [comment]
Thanks for your informations.
`ID_Einst_Popt_akt` is `1`, `ID_Einst_Popt_Nachlauf_akt` is `180` but `ID_Einst_P155_PumpHeat_Max` is `0`.
Any ideas why `ID_Einst_P155_PumpHeat_Max` is zero?

Can you see when the HUP is running? Exists a flag like the solar pump?

**Comment by Kars-de-Jong** [comment]
What is your `ID_Einst_P155_PumpHeatCtrl` set to? Mine is set to `1`, which displays as "Automatic" on the heat pump display.

Yes, there is a boolean for the HUP: `ID_WEB_HUPout`. You can also see the current control percentage, it's in `Circulation_Pump`.

**Comment by Kars-de-Jong** [comment]
If it is `None` it means this value is not present in your firmware version.
It probably also depends on your heatpump type.

**Comment by Kars-de-Jong** [comment]
Some of these parameters can only be set when logged in as installer by the way.

**Comment by pluskal** [comment]
@Kars-de-Jong, @BenPru this is extremely interesting conversation for me. Could you recommend some resources to read? 

I have very poor heating project that is missing parameters like `ΔT`.  My guess is that current `ΔT` is around `3 - 5 K`.

My goal is to further optimize my Alpha Innotec unit. It seems to me that is behaving very strangely.

The compressor is running on 20%. Is that healthy for the compressor to run for like 70-80% of the day? Or, during colder time periods to run non-stop on 20%? The high peeks around 12pm are hot water heating.

<img width="1092" alt="Screenshot 2022-11-20 at 11 15 33 AM" src="https://user-images.githubusercontent.com/3617487/202896616-4b953227-b3ea-46a5-b34d-014bf0e1eacc.png">

Each time the heating starts, pump is running for around 50% and than it slightly slows to 20%.

<img width="795" alt="Screenshot 2022-11-20 at 11 18 08 AM" src="https://user-images.githubusercontent.com/3617487/202896736-93690358-a4b3-4467-8e16-fd29364afb1f.png">

Additional very strange thing I have noticed is the speed of water circulation in the floor heating during the time that the heat pump is idle. It hits `1500-2000 l/min` almost constantly. 

<img width="1087" alt="Screenshot 2022-11-20 at 11 22 00 AM" src="https://user-images.githubusercontent.com/3617487/202896873-edc57832-9289-4dee-aebe-a2c9ca067da9.png">


It would be very interesting for me to see if the pump behaves similar for you as well, or if it is different to hear your thoughts on the topic.

**Comment by rhammen** [comment]
I'd say it looks pretty good! My (porbably oversimplified, but still) view:
- Inverter lifetime relates to start/stops --> less start/stops is better
- Heatpump efficiency is best with a low delta T. 
- On a day, with a certain outside temperature, you need to compensate a certain heatloss X of your house to outside. This heatloss is compensated by the heatpump. That could be both with a high(er) delta T and a shorter time, or with a low(er) delta T and a longer time. 
- That last is most efficient. And that is what your heatpump does! 
- So I'd say a constant frequency over the day is optimal.

My Alpha Innotec MSW4-16 behaves similarly, with yours still show more 'constant' behavior:

![Screenshot 2022-11-20 at 13-59-02 History – Home Assistant](https://user-images.githubusercontent.com/75572839/202903306-fd43e034-9b17-4d09-b1ba-d0d19b9c066a.png)

**Comment by BenPru** [comment]
@rhammen Yes I agree with you.
Before I controlled my luxtronik with home assistant my heatpump starts about 25 times per (cold) day and ran only 15 minutes every time. My passive house (is this right in englisch? - Passivhaus in german) needs only 12-15 kWh/m²/year and the underfloor heating says after a short time "enough".
The 15 minutes of heating was 5 minutes pump forerun, 5 minutes heating and 5 minutes pump forerun. Very inefficient.
With ha I detect the heatpump start and I increase the target correction to ensure a mininum of runtime.
Today I have normally only 2-4 impulses per day.

**Comment by pluskal** [comment]
Thanks @rhammen and @BenPru for the explanation.

I am still missing power measurement on the heat pump input so I am unable to compute COP. My current approximation is around 7.5 which seems to be a bit off to me.

My current settings are following:

<img width="807" alt="Screenshot 2022-11-20 at 6 10 23 PM" src="https://user-images.githubusercontent.com/3617487/202915710-14d75505-8b29-4823-8e8d-dc18067d8791.png">

I am not sure about `hysteresis`, but unless I lowered it down to `1.5K` the temperature inside was oscillating a lot. My goal is to have as constant temperature inside as possible.

My current temperatures are following:
<img width="522" alt="Screenshot 2022-11-20 at 6 15 46 PM" src="https://user-images.githubusercontent.com/3617487/202916041-42b9a21e-1065-424a-b31a-391f9e99ce74.png">

Is there a setting that allows to decrease `flow temperature`? To my best understanding, changing the heat curve sets the `return target`, so I tried to lower and flatten it which based on my observation prolonged the heating time and decreased frequency.

<img width="805" alt="Screenshot 2022-11-20 at 6 19 52 PM" src="https://user-images.githubusercontent.com/3617487/202916185-d79c623b-f60d-477c-ae49-1bc6e59a426f.png">

What is the minimal frequency your heat pump lowers to? If I remember correctly, with the previous firmware it was `15%`, now the minimum seems to be `20%`. I am thinking about updating to `V3.87.1-8454`.

**Comment by BenPru** [comment]
> I am still missing power measurement on the heat pump input so I am unable to compute COP.

As far as I know, Luxtronik only has an energy measurement for the output by default, but not for the input. I use ha glow to measure the input consumption at the utility electricity meter.

**Comment by Kars-de-Jong** [comment]
Haven’t seen it go below 20%, firmware version V3.85.9.

At 20% my compressor produced 1875 W thermal power today, and consumed 180 W electric power (compressor only) with the heat source coming in at 11,5 °C. Power for pumps and the controller was 43 W. So COP at that time was > 8.

**Comment by rhammen** [comment]
> Is there a setting that allows to decrease `flow temperature`? To my best understanding, changing the heat curve sets the `return target`, so I tried to lower and flatten it which based on my observation prolonged the heating time and decreased frequency.
> 
> <img alt="Screenshot 2022-11-20 at 6 19 52 PM" width="805" src="https://user-images.githubusercontent.com/3617487/202916185-d79c623b-f60d-477c-ae49-1bc6e59a426f.png">

@pluskal Wow, that is a really flat heating curve!
I'd say you must have a very well insulated home for this to work when it is cold outside!
On https://mnemotron.de/lux/heatcurve.html you can visualize the heating curve based on your settings. The diagram will show you on the Y-axis the target return temperature as a function of the outdoor temperature (X-axis).
Your heatcurve is really flat: at -20 C outside you set a target return temperature of 24.5 C, which is only about 1 degree higher than at 0 C outside. If your house is less well isolated, you loose more energy at -20 C than at 0 C, so you need a significantly higher return temperature.

I do not really understand why you set parameter 12 to 21.5 C if your home is well insulated.
21.5 degrees means that at outside temperature of +20C , you still want a return temperature of 21.5 C.
I'd set parameter 12 to 18 C  (or even lower if your house keeps its temperature at an outside temperature of 10-15 degrees.
And then set parameter 11 to the lowest value that keeps your home warm
Note that if you change parameter 12 by -3.5 C, you must increase parameter 11 by approx. +3.5 C to keep the endpoint the same. (in your case thus from 23.5 --> 27 C)

Note: for comparisio, my heatcurve uses 18C and 33C. I did not really spend time optimizing it.

**Comment by pluskal** [comment]
@rhammen, thanks for the detailed explanation! I really appreciate it!

> I'd say you must have a very well insulated home for this to work when it is cold outside!

It is hard to estimate precisely, but according to the project, my house should need less than `20 kWh/m²/year ` for heating. The house is well insulated but has a huge thermal capacity, so it reacts very slowly to outside temperature changes.

> I do not really understand why you set parameter 12 to 21.5 C if your home is well-insulated.

The reason is that the return temperature to drop below 20C` requires the room temperature to be less than `21C`. My optimal room temperature is `22.5C`. Based on empirical observation, it seemed like an appropriate setting, and the heating behaved desirably.

Because of the thermal capacity, it is not optimal to so heavily on the outside temperature. It is helpful to increase when the outside temperature is like `-20-10C` for a few days, but apart from that, I tried to have as low a heating temperature as possible to maximize COP. Moreover, it seemed better to have a meager and constant heat input that seemed to stabilize the inner room temperature.

The result is the following. The peek in 1NP (1 floor) happened because I ran a microwave to cook pizza for the launch. Also, the temperature stability is due to the low solar gain because there was a fog for almost a whole day.

<img width="348" alt="Screenshot 2022-11-22 at 6 30 14 PM" src="https://user-images.githubusercontent.com/3617487/203381916-f649011a-c683-4c67-bf28-e68cd052725b.png">


Is there a better way to achieve this?

Perhaps, I could flatten it even more to reduce the pump's time.

**Comment by Kars-de-Jong** [comment]
Found another source of electricity usage through the auxiliary power connection: the compressor heating.

It seems it turns on when the compressor temperature is below 20 degrees, and then heats it up to 25 degrees. Instantaneous power used by this is about 26 W.

I suppose this is meant to keep the oil warm enough for a compressor start.

**Comment by Kars-de-Jong** [comment]
Can be monitored by:
`calculations.ID_WEB_LIN_VDH` (compressor heating current temperature)
`calculations.ID_WEB_LIN_VDH_out` (`True` when compressor heating is on).

**Comment by BenPru** [comment]
> It seems it turns on when the compressor temperature is below 20 degrees, and then heats it up to 25 degrees. Instantaneous power used by this is about 26 W.

In my case it starts on <= 37 C° and stops >=39 C° every 15 minutes for 4 minutes and consumes about 65 W.
Any ideas howto stop this?

**Comment by Crashman1983** [comment]
Not possible. Needed to keep oil in liquid state.

Von meinem iPhone gesendet

Am 24.11.2022 um 17:17 schrieb BenPru ***@***.***>:

﻿

It seems it turns on when the compressor temperature is below 20 degrees, and then heats it up to 25 degrees. Instantaneous power used by this is about 26 W.

In my case it starts on <= 37 C° and stops >=39 C° every 15 minutes for 4 minutes and consumes about 65 W.
Any ideas howto stop this?

**Comment by BenPru** [comment]
But can I optimize it?
Your working range is 20-25 C ° and my 37-39. Why?

**Comment by Crashman1983** [comment]
I am using a AI LWDV - Air/Water Compressor and when it's cold outside it often shows me the message "heating up compressor". During this time (a few minutes each) the power consumption increases from 20W to about 45W. 

As far as I know this is normal behavior to protect the compressor and cannot be changed. But I think the high standby power consumption as well as power consumption for deicing is one of the main downsides of Alpha Innotec heat pumps. 

While during operation COP was about 5.7 - 6.0 today (at around 7°C), average COP (whole day, includes deicing) was only 5 (heat output per day / power consumption per day).

**Comment by Kars-de-Jong** [comment]
> But can I optimize it?
> Your working range is 20-25 C ° and my 37-39. Why?

Different compressor, different requirements? I have a WZSV 62K3M which uses R407C refrigerant.

**Comment by BenPru** [comment]
> high standby power consumption

:-(
In this heating period, my heat pump has only completed 18 impulses so far.
For example today I have `0 impulses` and `48 minutes solar pump` but `2,4 kWh` power consumption!

The circulation pump heating is running the hole day for nothing...
Also the compressor heater with 96 runs and about 6,5 hours runtime.

I'm considering creating a support ticket with Novelan or Alpha Innotec to clarify whether there isn't an optimization setting after all.

**Comment by pluskal** [comment]
> I'm considering creating a support ticket with Novelan or Alpha Innotec to clarify whether there isn't an optimization setting after all.

I am very curious about how that will go.  I tried to reach out to Alpha Innotec through their service partner in Czechia regarding a GDPR and the insecurity of their firmware... but no response at all.

**Comment by BenPru** [comment]
> I am very curious about how that will go. I tried to reach out to Alpha Innotec through their service partner in Czechia regarding a GDPR and the insecurity of their firmware... but no response at all.

My E-Mail to support@ait-deutschland.eu was answered one day ago.
Unfortunately, this did not bring any optimization approach.
Only now I know that the heating circulation pump also goes out in the heating off mode. If the outside temperature is below 5 C°, it stays on permanently.

**Comment by dickdahm** [comment]
> `ID_Einst_P155_PumpHeatCtrl` is `0`. Should I activate it? All `ID_Einst_P155_*` params are `0`.
> 
> `ID_WEB_HUPout` seems to work --> True/False changing. `Circulation_Pump` is always `None`.

Does anybody know what "P155_PumpHeatCtrl" does?
Did not see this before...

**Comment by dickdahm** [comment]
> Yesterday I activated my heatpump for the heating season. And now I read out the power usage of my heatpump with ha glow. I found out today that it has a big impact between heating "off" and "automatic" without running. The standby power usage in heating mode "off" is about ~13 Watt. In automatic mode the standby power usage is about 83 Watt and every 20 minutes it is ~150 Watt! Every 20 minutes the heatpump runs in mode "pump forerun" without doing something productive! In the past I thought this is a display bug in luxtronik but the usage show a real impact. My luxtronik version is V2.86.0 (Novelan LD7).
> 
> Has anyone something like this?

I just noticed a similar behaviour.
The compressor heater (Verdichter Heizung) is turning on several times a day. Maybe 20 times for exactly 5 minutes. The times when this happens is totally random.
Because of your interessting discussion here, I tried to disable pump optimazation, which if I understood you correctly, is responsible for this behaviour. The odd thing about this, is that this happens in summer at 35°C outside today, and domestic water and heating turned off...

**Comment by Crashman1983** [comment]
The compressor heater is heating oil. This needs to be at above 40 degree Celsius afaik.

Power consumption is rising to 45W/h in this case. Without, standby power for luxtronik and everything else is about 20W/h.

this is as designed and cannot be changed. Unfortunately it is too much.

**Comment by BenPru** [comment]
> Power consumption is rising to 45W/h in this case. Without, standby power for luxtronik and everything else is about 20W/h.

Yes, this is not very good. What happens if the power is down for hours or days?
In my case the normal consumption is 12W/h and with Compressor heater about 85W/h! And my heatpump has 3/4 the year only to control the solarpump! My heatpump needs about 1,2 kW/h per day without solarpump or heating something!
